File: objc/Mediator/Mediator/Button.m
#import "Button.h"

@implementation Button

-(instancetype)initWithMediator:(id<DialogBoxMediator>)med {
    if(self = [super initWithMediator:med]) {
        [super initWithMediator:med];
        _title = @"";
        _enabledState = YES;
    }
}

-(void)setTitle:(NSString*)title {
    _title = title;
}

-(void)setEnabledState:(BOOL)state {
    _enabledState = state;
}

-(NSString*)toString {
    return [[NSString alloc] initWithFormat:@"Title: %@, State: %@", _title, _enabledState];
}

@end

File: swift/Mediator/Mediator/Button.swift
import Foundation

public class Button : Widget {
    var title: String = ""
    var enabledState: Bool = true
    
    func setTitle(_ title: String) -> Void {
        self.title = title
    }
    
    func setEnabledState(_ state: Bool) -> Void {
        self.enabledState = state
    }
    
    func toString() -> String {
        return String("Title: \(self.title), State: \(self.enabledState)")
    }
}
